### PR TITLE
Fixed: Monsters stepdancing and place creature in field 

### PR DESCRIPTION
**Pull request by dex-89**
- **Monsters can now stepdancing in fields when attacking a foe.**
- **Monsters are now ignoring fields in the ground when they are created.**

**Author:** @FakeShinoda

### CODE REVIEW

File: src/monster.cpp [hunk 1]
```diff
@@ -1886,7 +1886,7 @@ bool Monster::canWalkTo(Position pos, Direction moveDirection) const
 		}
 
 		Tile* walkTile = g_game.map.getTile(pos);
-		if (walkTile && walkTile->getTopVisibleCreature(this) == nullptr && walkTile->queryAdd(0, *this, 1, FLAG_PATHFINDING) == RETURNVALUE_NOERROR) {
+		if (tile && tile->getTopVisibleCreature(this) == nullptr && tile->queryAdd(0, *this, 1, FLAG_PATHFINDING | FLAG_IGNOREFIELDDAMAGE) == RETURNVALUE_NOERROR) {
```
Comment (gerotib) [inline src/monster.cpp:1889]:
```suggestion
		if (walkTile && walkTile->getTopVisibleCreature(this) == nullptr && walkTile->queryAdd(0, *this, 1, FLAG_PATHFINDING | FLAG_IGNOREFIELDDAMAGE) == RETURNVALUE_NOERROR) {
```
should not be changed

Comment (FakeShinoda) [reply src/monster.cpp:1889]: replace walkTile to tile

File: src/monster.cpp [hunk 2]
```diff
@@ -1886,7 +1886,7 @@ bool Monster::canWalkTo(Position pos, Direction moveDirection) const
 		}
 
 		Tile* walkTile = g_game.map.getTile(pos);
```
Comment (gerotib) [inline src/monster.cpp]:
```suggestion
		Tile* tile = g_game.map.getTile(pos);
```

**Comment by gerotib** [comment]
Monsters are walking on fields all the time with this change, not only while dancing.

**Comment by dex-89** [comment]
Maybe this: https://otland.net/threads/monster-walk-field.246167/ will help something?

**Comment by FakeShinoda** [comment]
> 
> 
> Monsters are walking on fields all the time with this change, not only while dancing.

Right, thanks for the report. <3

File: src/map.cpp [hunk 3]
```diff
@@ -215,7 +227,10 @@ bool Map::placeCreature(const Position& centerPos, Creature* creature, bool exte
 
 		if (!foundTile) {
 			return false;
-		}
+		} else {
+			if (monster) {
+				monster->ignoreFieldDamage = false;
+			}
```
Comment (gerotib) [inline src/map.cpp:233]:
```suggestion
			}
		}
```
Missing here.

**Review by gerotib — COMMENTED** [review]
With this small fix this pr seems to work fine.

**Review by gerotib — APPROVED** [review]

**Review by FakeShinoda — APPROVED** [review]

**Pull request merged**